dsf-mpi/dsf-mpi-client-pdq/src/main/java/org/highmed/mpi/client/message/MessageHelper.java

```diff
@@ -66,10 +66,19 @@ public QBP_Q21 createPatientDemographicsQuery(String senderApplication, String s
 
 		Terser terser = new Terser(qbp_q21);
 
-		for (int i = 0; i < searchParameters.size(); i++)
+		int qpd3counter = 0;
+		for (QueryParameter searchParameter : searchParameters)
 		{
-			terser.set("/QPD-3(" + i + ")-1", searchParameters.get(i).getField());
-			terser.set("/QPD-3(" + i + ")-2", searchParameters.get(i).getValue());
+			if (searchParameter.hasLocation())
+			{
+				terser.set(searchParameter.getLocation(), searchParameter.getValue());
+			}
+			else
+			{
+				terser.set("/QPD-3(" + qpd3counter + ")-1", searchParameter.getField());
+				terser.set("/QPD-3(" + qpd3counter + ")-2", searchParameter.getValue());
+				qpd3counter++;
+			}
 		}
```

Comment:
The qpd3counter variable does not need to be visible outside the loop. I would suggest using a `for (int`... loop.
```java
for (int qpd3counter = 0; qpd3counter < searchParameters.size(); qpd3counter++)
{
	QueryParameter searchParameter = searchParameters.get(qpd3counter);

	if (searchParameter.hasLocation())
	{
		terser.set(searchParameter.getLocation(), searchParameter.getValue());
	}
	else
	{
		terser.set("/QPD-3(" + qpd3counter + ")-1", searchParameter.getField());
		terser.set("/QPD-3(" + qpd3counter + ")-2", searchParameter.getValue());
	}
}
```